ios/FluentUI/People Picker/ContactCollectionView.swift

```diff
@@ -0,0 +1,162 @@
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    /// The array of PersonaData which is used to create each ContactView.
+    /// The height constraint of the ContactCollectionView is updated when the count increases from 0 or decreases to 0.
+    @objc public var contactList: [PersonaData] = [] {
+        didSet {
+            if (oldValue.count == 0 && contactList.count > 0) || (oldValue.count > 0 && contactList.count == 0) {
+                setupHeightConstraint()
+            }
+        }
+    }
+
+    open weak var contactCollectionViewDelegate: ContactCollectionViewDelegate?
+
+    private func configureCollectionView() {
+        translatesAutoresizingMaskIntoConstraints = false
+        showsHorizontalScrollIndicator = false
+        showsVerticalScrollIndicator = false
+        backgroundColor = Colors.surfacePrimary
+        dataSource = self
+        contentInset = UIEdgeInsets(top: 0, left: Constants.leadingInset, bottom: 0, right: 0)
+    }
+
+    @objc private func setupHeightConstraint() {
+        let height = (contactList.count > 0) ? UIApplication.shared.preferredContentSizeCategory.contactHeight : 0
+        heightConstraint.constant = height
+    }
+
+    /// If the Contact is not fully visible in the scroll frame, scrolls the view by an offset large enough
+    /// so that the tapped Contact is fully visible as well as 20px of the next Contact.
+    private func scrollToContact(at indexPath: IndexPath) {
+        guard let cell = cellForItem(at: indexPath) else {
+            return
+        }
+
+        let cellWidth = cell.frame.width
+        let cellFrame = cell.contentView.convert(cell.bounds, to: self)
+        let cellLeftPosition = cellFrame.origin.x
+        let cellRightPosition = cellLeftPosition + cellWidth
+        let viewLeadingPosition = bounds.origin.x
+        let viewTrailingPosition = viewLeadingPosition + frame.size.width
+
+        let extraScrollWidth = layout.minimumLineSpacingForSectionAt(section: 0) + Constants.amountOfNextContactToShow
+        var offSet = contentOffset.x
+        if cellLeftPosition < viewLeadingPosition {
+            offSet = cellLeftPosition - extraScrollWidth
+            offSet = max(offSet, -Constants.leadingInset)
+        } else if cellRightPosition > viewTrailingPosition {
+            let maxOffsetX = contentSize.width - frame.size.width + extraScrollWidth
+            offSet = cellRightPosition - frame.size.width + extraScrollWidth
+            offSet = min(offSet, maxOffsetX)
+        }
+
+        if offSet != contentOffset.x {
+            setContentOffset(CGPoint(x: offSet, y: contentOffset.y), animated: true)
+        }
+    }
+
+    private struct Constants {
+        static let leadingInset: CGFloat = 16.0
+        static let amountOfNextContactToShow: CGFloat = 20.0
+    }
+
+    private let layout: ContactCollectionViewLayout
+    private var widthConstraint: NSLayoutConstraint?
```

Comment:
Is widthConstraint being used anywhere? Can't seem to find any usage.

Reply:
You're correct, it's not being used anywhere. I'll remove it. It's weird that I tried using the "Static Analyzer" in Xcode which is supposed to show unused variables, but it didn't report it as being unused.

https://developer.apple.com/library/archive/documentation/ToolsLanguages/Conceptual/Xcode_Overview/AnalyzingYourCode.html

> Dead store (unused variable) flaws

Reply:
Discussed offline with Tom and it appears that it isn't able to discover unused **member** variables.